torchdynamo/skipfiles.py

```diff
@@ -104,6 +104,16 @@ def _module_dir(m: types.ModuleType):
 FILENAME_ALLOWLIST = {
     torch.nn.Sequential.__init__.__code__.co_filename,
 }
+
+# Include optimizer code for tracing
+FILENAME_ALLOWLIST |= set(
+    [
+        inspect.getfile(obj)
+        for obj in torch.optim.__dict__.values()
+        if inspect.isclass(obj)
```

Comment:
nit - you do this iteration twice, make a helper?

Reply:
The second iteration only contains 8 values, I don't think this is worth it, it's also only run once when this file is imported.